### PR TITLE
Adding MonoT5 finetuning script on MS MARCO GPU

### PR DESCRIPTION
**Pull request by vjeronymo2**
More information on experiments-finetune-monot5-gpu.md.

### CODE REVIEW

File: pygaggle/run/monot5_reranker.py [hunk 1]
```diff
@@ -0,0 +1,69 @@
+import argparse
```
Comment (rodrigonogueira4) [inline pygaggle/run/monot5_reranker.py:1]: Perhaps call this file "evaluate_monot5_reranker.py"?

File: docs/experiments-finetune-monot5-gpu.md [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+```
+nohup python -um pygaggle.run.finetune_monot5 --triples_path triples/triples.train.small.tsv \
+                                              --save_every_n_steps 10000 \
+                                              --output_model_path monoT5_model/ &
+tail -f nohup
+```
+Here, we consider one epoch to be 6.4e5 of lines of the triples.train.small.tsv, which corresponds to 12.8e5 training examples (positives + negatives) and 10k steps using batch of 128. The number of epochs defaults to 10 (100k steps). Adding the "--epoch" argument with an integer will include `n * 6.4e5` lines to the training set.
+If you don't wish to save checkpoints every N step, simply omit the --save_every_n_steps argument.
+
+With a Nvidia V100 it takes around 2~3 days to train the model on 10 epoch.
```
Comment (rodrigonogueira4) [inline docs/experiments-finetune-monot5-gpu.md:50]:
```suggestion
With an Nvidia V100, it takes around 2~3 days to train the model for 10 epochs.
```

File: docs/experiments-finetune-monot5-gpu.md [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+```
+mkdir monoT5_model/
+```
+### Train
+To train the MonoT5 model, run the following command:
+```
+nohup python -um pygaggle.run.finetune_monot5 --triples_path triples/triples.train.small.tsv \
+                                              --save_every_n_steps 10000 \
+                                              --output_model_path monoT5_model/ &
+tail -f nohup
```
Comment (rodrigonogueira4) [inline docs/experiments-finetune-monot5-gpu.md:45]: It should be nohup**.out**?

**Review by rodrigonogueira4 — COMMENTED** [review]
Awesome, thanks for adding this PR. Just a few minor comments

File: pygaggle/run/monot5_reranker.py (hunk 1 above)
Comment (vjeronymo2) [reply pygaggle/run/monot5_reranker.py]: Done

File: pygaggle/run/finetune_monot5.py [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+
+    train_args = Seq2SeqTrainingArguments(
+        output_dir=args.output_model_path,
+        do_train=True,
+        save_strategy=strategy,
+        save_steps =steps, 
+        logging_steps=100,
+        per_device_train_batch_size=8,
+        gradient_accumulation_steps=16,
+        learning_rate=3e-4,
```
[the remark this answers is not in the document]
Comment (vjeronymo2) [reply pygaggle/run/finetune_monot5.py]: Done

**Review by rodrigonogueira4 — APPROVED** [review]
Awesome, thanks for implementing this!

**Pull request merged**